tests/testthat/test-data-grid.R

```diff
@@ -12,3 +12,18 @@ test_that("can generate typical values", {
   expect_equal(out$x, c("a", "a", "b", "b"))
   expect_equal(out$y, c("a", "b", "a", "b"))
 })
+
+test_that("data_grid() works as expected when no model is supplied", {
+  expect_equal(
+    tidyr::expand(mtcars, cyl = seq_range(cyl, 9)),
+    data_grid(mtcars, cyl = seq_range(cyl, 9))
+  )
+})
+
+test_that("data_grid() returns a tibble", {
+  mod <- lm(mpg ~ wt + cyl + vs, data = mtcars)
```

Comment:
Since you're testing `data_grid`, you should make the model as simple as possible.